### ISSUE TITLE
Using expect_stdout() with multiline output

### ISSUE THREAD

**Issue by jdblischak** [markvanderloo/tinytest#58]
Thanks for the great package. I'm trying to use `expect_stdout()` to test the output from S3 methods I've written like `print()` and `summary()`, and it has been more difficult than I had anticipated.

Below are examples. If the output is one line, the test passes. If the output is more than one line, the test produces a warning. And instead of returning a single `TRUE` or `FALSE`, it returns a logical value for each line.

```
> packageVersion("tinytest")
[1] ‘1.2.2’
> library(tinytest)
> expect_stdout(
+   str(1:10),
+   "int"
+ )
----- PASSED      : <-->
 call| expect_stdout(str(1:10), "int") 

> expect_stdout(
+   str(mtcars),
+   "carb"
+ )
 
Warning messages:
1: In if (!result) msg <- sprintf("output '%s'\n does not match pattern '%s'",  :
  the condition has length > 1 and only the first element will be used
2: In if (result) NA_character_ else "xcpt" :
  the condition has length > 1 and only the first element will be used
3: In if (is.na(x)) sprintf("SIDEFX[%s]", short) :
  the condition has length > 1 and only the first element will be used
```

My workaround was to write a helper function that combines all of the output into a single string.

```
concatenateOutput <- function(x) {
  output <- utils::capture.output(print(x))
  output <- paste(output, collapse = "\n")
  return(output)
}
```

However, this fails for some reason even though it works with `grepl()`.

```
> concatenateOutput(str(mtcars))
[1] "'data.frame':\t32 obs. of  11 variables:\n $ mpg : num  21 21 22.8 21.4 18.7 18.1 14.3 24.4 22.8 19.2 ...\n $ cyl : num  6 6 4 6 8 6 8 4 4 6 ...\n $ disp: num  160 160 108 258 360 ...\n $ hp  : num  110 110 93 110 175 105 245 62 95 123 ...\n $ drat: num  3.9 3.9 3.85 3.08 3.15 2.76 3.21 3.69 3.92 3.92 ...\n $ wt  : num  2.62 2.88 2.32 3.21 3.44 ...\n $ qsec: num  16.5 17 18.6 19.4 17 ...\n $ vs  : num  0 0 1 1 0 1 0 1 1 1 ...\n $ am  : num  1 1 1 0 0 0 0 0 0 0 ...\n $ gear: num  4 4 4 3 3 3 3 4 4 4 ...\n $ carb: num  4 4 1 1 2 1 4 2 2 4 ...\nNULL"

> grepl("disp", concatenateOutput(str(mtcars)))
[1] TRUE

> expect_stdout(
+   concatenateOutput(str(mtcars)),
+   "disp"
+ )
----- FAILED[xcpt]: <-->
 call| expect_stdout(concatenateOutput(str(mtcars)), "disp")
 diff|  
```

Is there a different strategy I need to use? I looked at the example tests for `expect_stdout()`, but they all test single line output.

https://github.com/markvanderloo/tinytest/blob/fb91fbc8001d31162df5e4ac7262da280b60b51e/pkg/inst/tinytest/test_tiny.R#L67-L70

Would it be possible to extend `expect_stdout()` to handle multiline output, similar to how `expect_message()` handles multiline messages (#51)?

**Comment by markvanderloo** [comment]
Thanks for the report John! I fixed it in the current dev version, see https://github.com/markvanderloo/tinytest/commit/6a0215a35ec8b9fb0a1f2627f2eaa737ee62f985

As a token of my gratitude, you have been immortalized in the package's [NEWS](../blob/master/pkg/NEWS) file.

**Comment by jdblischak** [comment]
> Thanks for the report John! I fixed it in the current dev version, see 6a0215a

Wonderful! Thanks for the quick fix. I confirmed the dev version can handle my multiline output tests.

> As a token of my gratitude, you have been immortalized in the package's NEWS file.

I'm honored :-)
